### DISCUSSION TITLE
Connect-ExchangeOnline and C# microsoft.powershell.sdk library

### DISCUSSION THREAD

**Discussion by ayblanchet** [PowerShell/PowerShell#18518] [Q&A]
Hello,

I try to use Connect-ExchangeOnline in C# azure function using microsoft.powershell.sdk.
But when I try to call Connect-ExchangeOnline I have the error "The specified module 'ExchangeOnlineManagement' was not loaded because no valid module file was found in any module directory."
Impossible to install or import ExchangeOnlineManagement module, Install-Module is not found too. 
Can you help me, and say me what is missing to get ExchangeOnlineManagement module?
I use these packages:
        <PackageReference Include="Microsoft.PowerShell.Commands.Diagnostics" Version="7.2.7" />
	<PackageReference Include="Microsoft.PowerShell.SDK" Version="7.2.7" />
        <PackageReference Include="Microsoft.WSMan.Management" Version="7.2.7" />

Best Regards,

**Comment by jborean93** [comment]
The SDK is purely PowerShell itself whereas cmdlets like `Connect-ExchangeOnline` come from modules, same with `Install-Module`. I'm unsure how Azure Functions work and whether it is possible to also bundle modules manually but you need to look further into that.
